dataland-qa-service/src/main/kotlin/org/dataland/datalandqaservice/controller/QaController.kt

```diff
@@ -0,0 +1,101 @@
+package org.dataland.datalandqaservice.controller
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.dataland.datalandbackendutils.exceptions.InvalidInputApiException
+import org.dataland.datalandbackendutils.model.QAStatus
+import org.dataland.datalandmessagequeueutils.cloudevents.CloudEventMessageHandler
+import org.dataland.datalandmessagequeueutils.constants.ExchangeNames
+import org.dataland.datalandmessagequeueutils.constants.MessageType
+import org.dataland.datalandmessagequeueutils.constants.RoutingKeyNames
+import org.dataland.datalandmessagequeueutils.messages.QaCompletedMessage
+import org.dataland.datalandqaservice.api.QaApi
+import org.dataland.datalandqaservice.entities.DatasetReviewStatusEntity
+import org.dataland.datalandqaservice.repositories.DatasetReviewStatusRepository
+import org.dataland.keycloakAdapter.auth.DatalandAuthentication
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.ResponseEntity
+import org.springframework.transaction.annotation.Transactional
+import org.springframework.web.bind.annotation.RestController
+import kotlin.jvm.optionals.getOrElse
+
+/**
+ * Controller for the QA service API
+ */
+@RestController
+class QaController(
+    @Autowired val datasetReviewStatusRepository: DatasetReviewStatusRepository,
+    @Autowired var cloudEventMessageHandler: CloudEventMessageHandler,
+    @Autowired var objectMapper: ObjectMapper,
+) : QaApi {
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    @Transactional
+    override fun getUnreviewedDatasets(): ResponseEntity<List<String>> {
+        logger.info("Received request to respond with IDs of unreviewed datasets")
+        return ResponseEntity.ok(datasetReviewStatusRepository.getSortedPendingDataIds())
+    }
+
+    @Transactional
+    override fun assignQualityStatus(dataId: String, qualityStatus: QAStatus, comment: String?):
+        ResponseEntity<Nothing> {
+        logger.info("Assigning quality status ${qualityStatus.name} to dataset with ID $dataId")
+        if (qualityStatus == QAStatus.Pending) {
+            throw InvalidInputApiException(
+                "Quality \"Pending\" cannot be assigned to a reviewed dataset",
+                "Quality \"Pending\" cannot be assigned to a reviewed dataset",
```

Comment:
I think it makes more sense to do the check first, and then log the message. What do you think?

Reply:
To be pushed

Reply:
Yes good idea.  Log the "tryting to..." stuff and then the other thing after check.

Then in the future we can exactly track what happened by looking at the logs.